### PR TITLE
Remove SP.synPermTrim

### PR DESCRIPTION
**Pull request by ctrl-z-9000-times**
Pull request for issue #150

### CODE REVIEW

File: src/nupic/algorithms/SpatialPooler.cpp [hunk 1]
```diff
@@ -596,18 +584,16 @@ vector<Real> SpatialPooler::initPermanence_(const vector<UInt> &potential, //TOD
     } else {
       perm[i] = initPermNonConnected_();
     }
-    perm[i] = perm[i] < synPermTrimThreshold_ ? 0.0f : perm[i];
```
[the remark this answers is not in the document]
Comment (breznak) [reply src/nupic/algorithms/SpatialPooler.cpp]: ok, not even an issue, realizing this is only used in SP::init !

**Comment by breznak** [comment]
fine with me :+1: 
I'll wait if @dkeeney has strong objections to the small API cleanup, otherwise merge. 
@ctrl-z-9000-times I've noticed you're preparing some work for #93 in connections branch :wink: , when you are ready, please share it rather earlier than later..I may have some comments to it.

File: src/nupic/algorithms/SpatialPooler.cpp [hunk 2]
```diff
@@ -596,18 +584,16 @@ vector<Real> SpatialPooler::initPermanence_(const vector<UInt> &potential, //TOD
...
 }
 
 
-void SpatialPooler::clip_(vector<Real> &perm, bool trim) const {
-  const Real minVal = trim ? synPermTrimThreshold_ : synPermMin_;
+void SpatialPooler::clip_(vector<Real> &perm) const {
   for (auto &elem : perm) {
     elem = elem > synPermMax_ ? synPermMax_ : elem; //crop upper bound
-    elem = elem < minVal ? synPermMin_ : elem; //crop lower
+    elem = elem < synPermMin_ ? synPermMin_ : elem; //crop lower
```
Comment (breznak) [inline src/nupic/algorithms/SpatialPooler.cpp]: nit, replace these short-hand if-then-else, with only if(e < ..) { };

**Comment by ctrl-z-9000-times** [comment]
Ok, i will make these changes next chance i get, probable tomorrow morning. I live in EST btw

**Pull request merged**